Source/WebCore/platform/graphics/ca/TileController.cpp

```diff
@@ -743,12 +849,14 @@ void TileController::updateTileCoverageMap()
 
 IntRect TileController::tileGridExtent() const
 {
-    return tileGrid().extent();
+    const TileGrid& grid = m_asyncTileConfigurationChangeData ? *m_asyncTileConfigurationChangeData->pendingTileGrid : tileGrid();
+    return grid.extent();
 }
 
 double TileController::retainedTileBackingStoreMemory() const
 {
-    double bytes = tileGrid().retainedTileBackingStoreMemory();
+    const TileGrid& grid = m_asyncTileConfigurationChangeData ? *m_asyncTileConfigurationChangeData->pendingTileGrid : tileGrid();
```

Comment:
All of this should get rolled into `TileController::tileGrid()`

Reply:
i.e. `m_asyncTileConfigurationChangeData ? *m_asyncTileConfigurationChangeData->pendingTileGrid : tileGrid();`